packages/react-devtools-shared/src/__tests__/editing-test.js

```diff
@@ -0,0 +1,1103 @@
+        });
+        jest.runOnlyPendingTimers();
+      }
+
+      overrideState(['new'], 'value');
+      expect(committedState).toStrictEqual({
+        array: [1, 2, 3],
+        object: {
+          nested: 'initial',
+        },
+        shallow: 'initial',
+        new: 'value',
+      });
+
+      overrideState(['object', 'new'], 'value');
+      expect(committedState).toStrictEqual({
+        array: [1, 2, 3],
+        object: {
+          nested: 'initial',
+          new: 'value',
+        },
+        shallow: 'initial',
+        new: 'value',
+      });
+
+      overrideState(['array', 3], 'new value');
+      expect(committedState).toStrictEqual({
+        array: [1, 2, 3, 'new value'],
+        object: {
+          nested: 'initial',
+          new: 'value',
+        },
+        shallow: 'initial',
+        new: 'value',
+      });
+    });
+
+    it('should have deletable keys', async () => {
+      await mountTestApp();
+
+      function deletePath(path) {
+        const rendererID = utils.getRendererID();
+        bridge.send('deletePath', {
+          id,
+          path,
+          rendererID,
+          type: 'state',
+        });
+        jest.runOnlyPendingTimers();
+      }
+
+      deletePath(['shallow']);
+      expect(committedState).toStrictEqual({
+        array: [1, 2, 3],
+        object: {
+          nested: 'initial',
+        },
+      });
+
+      deletePath(['object', 'nested']);
+      expect(committedState).toStrictEqual({
+        array: [1, 2, 3],
+        object: {},
+      });
+
+      deletePath(['array', 1]);
+      expect(committedState).toStrictEqual({
+        array: [1, 3],
+        object: {},
+      });
+    });
+  });
+
+  describe('hooks', () => {
+    let committedState;
+    let hookID;
+    let id;
+
+    async function mountTestApp() {
+      function FunctionComponent() {
+        const [state] = React.useState({
+          array: [1, 2, 3],
+          object: {
+            nested: 'initial',
+          },
+          shallow: 'initial',
+        });
+        React.useLayoutEffect(() => {
+          committedState = state;
+        });
+        return null;
+      }
+
+      const container = document.createElement('div');
+      await utils.actAsync(() =>
+        ReactDOM.render(<FunctionComponent />, container),
+      );
+
+      hookID = 0; // index
+      id = ((store.getElementIDAtIndex(0): any): number);
+
+      expect(committedState).toStrictEqual({
+        array: [1, 2, 3],
+        object: {
+          nested: 'initial',
+        },
+        shallow: 'initial',
+      });
+    }
+
+    it('should have editable values', async () => {
+      await mountTestApp();
+
+      function overrideHookState(path, value) {
+        const rendererID = utils.getRendererID();
+        bridge.send('overrideValueAtPath', {
+          hookID,
+          id,
+          path,
+          rendererID,
+          type: 'hooks',
+          value,
+        });
+        jest.runOnlyPendingTimers();
```

Comment:
May be good to add a comment to explain what timer we're ticking here (or a helper function like `flushPendingUpdate`)